### PR TITLE
home sign in component

### PR DESCRIPTION
**Pull request by adammeyer**
For the AEM home page

### CODE REVIEW

File: src/app/homeSignIn/homeSignIn.component.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+class HomeSignInController {
+
+  /* @ngInject */
+  constructor(sessionService, sessionModalService) {
+    this.sessionService = sessionService;
+    this.sessionModalService = sessionModalService;
+  }
+
+  $onInit() {
```
Comment (OzzieOrca) [inline src/app/homeSignIn/homeSignIn.component.js]: Can you test this? I think it'd be pretty easy. You might have to mock sessionService.

File: src/app/homeSignIn/homeSignIn.component.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        this.isSigningIn = false;
+        this.showSignInForm = false;
+      }, ( error ) => {
+        this.isSigningIn = false;
+        this.errorMessage = error.data.error;
+      } );
+  }
+
+  signUp() {
+    this.sessionModalService.signUp();
```
Comment (OzzieOrca) [inline src/app/homeSignIn/homeSignIn.component.js]: Can you test this too? I think it would just be a single spy to check if it has been called.

**Comment by OzzieOrca** [comment]
Looks good, I'm just curious, where this will be used? Is this in the navbar or part of a panel somewhere? I guess idk what the home page looks like.

**Comment by adammeyer** [comment]
It's going to be the sign in part about half way down on the home page.  http://crugive.site/homepage/

**Comment by Omicron7** [comment]
👍

File: src/app/homeSignIn/homeSignIn.spec.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  describe( 'signUp', () => {
+    let deferred;
+    beforeEach( inject( function ( _$q_ ) {
+      deferred = _$q_.defer();
+      spyOn( $ctrl.sessionModalService, 'signUp' ).and.callFake( () => deferred.promise );
+      $ctrl.signUp();
+    } ) );
+
+    it( 'opens signUp Modal', () => {
+      expect( $ctrl.sessionModalService.signUp ).toHaveBeenCalled();
```
Comment (OzzieOrca) [inline src/app/homeSignIn/homeSignIn.spec.js]: I don't think you need the calling strategy if you're only testing that it has been called and can leave it as `spyOn( $ctrl.sessionModalService, 'signUp' )`.

**Comment by OzzieOrca** [comment]
The home page is cool. Hadn't seen it yet.
👍

**Pull request merged**